tests/typecheck/good/type-subst-avoids-capture-2.scilla

```diff
@@ -0,0 +1,4 @@
+let id_with_dummy = tfun 'A => tfun 'B => fun (x : 'A) => x in
```

Comment:
This test didn't fail previously because `free_tvars` used to return `['Y]` for `'X -> 'Y`. So I added this for symmetry.